File: eecs168/2020fall/TR/2020.11.04.classes-intro/Triangle-example/Triangle.cpp
//Triangle.cpp
#include "Triangle.h"
//Define the member methods
//for the Triangle class

double Triangle::area()
{
	//This method has
	//access to all members
	//of the Triangle class
	double ans = 0;
	ans = 0.5*base*height;
	return (ans);
}

File: eecs168/2025summer/2025.07.15.classes-part-2/triangle.py
#triangle.py

class Triangle:
    def __init__(self):
        print('__init__ called')
        #member variables
        self.base = -1
        self.height = -1

    #Goal: Define an area method
    def area(self):
        ans = (1/2)*self.base*self.height
        return ans
